reconcile/aws_cloudwatch_log_retention/integration.py

```diff
@@ -64,51 +65,75 @@ def check_cloudwatch_log_group_tag(
-                        log_group_tags = log_group["tags"]
-                        if not all(
-                            item in log_group_tags.items()
-                            for item in MANAGED_TAG.items()
-                        ):
-                            logging.info(
-                                f"Setting tag {MANAGED_TAG} for group {group_name}"
-                            )
-                            if not dry_run:
-                                awsapi.create_cloudwatch_tag(
-                                    aws_acct, group_name, MANAGED_TAG
-                                )
-                        if retention_days != int(
-                            cloudwatch_cleanup_entry.log_retention_day_length
-                        ):
-                            logging.info(
-                                f" Setting {group_name} retention days to {cloudwatch_cleanup_entry.log_retention_day_length}"
-                            )
-                            if not dry_run:
-                                awsapi.set_cloudwatch_log_retention(
-                                    aws_acct,
-                                    group_name,
-                                    int(
-                                        cloudwatch_cleanup_entry.log_retention_day_length
-                                    ),
-                                )
+    app_interface_aws_accounts = get_aws_accounts(cleanup=True)
+    with create_awsapi_client(app_interface_aws_accounts, 1) as awsapi:
+        for app_interface_aws_acct in app_interface_aws_accounts:
+            aws_act_name = app_interface_aws_acct["name"]
+            cloudwatch_cleanup_list = []
+            if app_interface_aws_acct.get("cleanup"):
+                cloudwatch_cleanup_list = get_app_interface_cloudwatch_retention_period(
+                    app_interface_aws_acct
+                )
+            aws_log_group_list = []
+            try:
+                aws_log_group_list = get_log_group_list(awsapi, app_interface_aws_acct)
+            except ClientError as e:
+                if e.response["Error"]["Code"] == "AccessDeniedException":
+                    logging.info(f" Access denied for {aws_act_name}. Skipping...")
+                    continue
+            for aws_log_group in aws_log_group_list:
+                group_name = aws_log_group["logGroupName"]
+                retention_days = aws_log_group.get("retentionInDays")
+
+                cloudwatch_cleanup_entry = next(
+                    (
+                        c
+                        for c in cloudwatch_cleanup_list
+                        if c.log_regex.match(group_name)
+                    ),
+                    None,
+                )
+                log_group_tags = aws_log_group["tags"]
+                if not all(
+                    item in log_group_tags.items() for item in MANAGED_TAG.items()
+                ):
```

Comment:
Do you mean `if log_group_tags.get("managed_by_integration") != QONTRACT_INTEGRATION:` here?